addons/binding/org.openhab.binding.elkm1/src/main/java/org/openhab/binding/elkm1/handler/ElkM1BridgeHandler.java

```diff
@@ -0,0 +1,361 @@
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to open socket to alarm");
+        }
+    }
+
+    /**
+     * Called when the configuration is updated. We will reconnect to the elk at this point.
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+        this.connection.removeElkListener(this);
+        this.connection.shutdown();
+        this.connection = new ElkAlarmConnection(getConfigAs(ElkAlarmConfig.class), messageFactory);
+        connection.addElkListener(this);
+        if (connection.initialize()) {
+            connection.sendCommand(new Version());
+            connection.sendCommand(new ZoneDefinition());
+            connection.sendCommand(new ZonePartition());
+            connection.sendCommand(new ZoneStatus());
+            connection.sendCommand(new ArmingStatus());
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Requesting version from alarm");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to open socket to alarm");
+        }
+    }
+
+    /**
+     * Shutdown the bridge.
+     */
+    @Override
+    public void dispose() {
+        connection.shutdown();
+        areas = null;
+        connection = null;
+        messageFactory = null;
+        assert (listeners.isEmpty());
+        super.dispose();
+    }
+
+    /**
+     * Handlers an incoming message from the elk system.
+     *
+     * @param message The message from the ekl to handle
+     */
+    @Override
+    public void handleElkMessage(ElkMessage message) {
+        logger.debug("Got elk message {}", message.toString());
+        if (message instanceof VersionReply) {
+            VersionReply reply = (VersionReply) message;
+            // Set the property.
+            getThing().setProperty(ElkM1BindingConstants.PROPERTY_VERSION, reply.getElkVersion());
+        }
+        if (message instanceof ZoneStatusReply) {
+            ZoneStatusReply reply = (ZoneStatusReply) message;
+            for (int i = 0; i < ElkMessageFactory.MAX_ZONES; i++) {
+                Thing thing = getThingForType(ElkTypeToRequest.Zone, i + 1);
+                if (thing != null) {
+                    ElkM1ZoneHandler handler = (ElkM1ZoneHandler) thing.getHandler();
+                    handler.updateZoneConfig(reply.getConfig()[i], reply.getStatus()[i]);
+                }
+            }
+        }
+        if (message instanceof ZonePartitionReply) {
+            ZonePartitionReply reply = (ZonePartitionReply) message;
+            for (int i = 0; i < ElkMessageFactory.MAX_ZONES; i++) {
+                Thing thing = getThingForType(ElkTypeToRequest.Area, reply.getAreas()[i]);
+                if (thing == null && !areas[reply.getAreas()[i] - 1]) {
```

Comment:
When this code gets an unconfigured zone (such as if zone numbers were skipped), the area returned is 0.  Subtracting 1 from it results in an index out of bounds exception.   Suggest adding a !=0 check into if statement
```
if (thing == null && reply.getAreas()[i] != 0 && !areas[reply.getAreas()[i] - 1])
```